api/v1beta2/flowcollector_types.go

```diff
@@ -0,0 +1,769 @@
+	// only endConnection events or "ALL" to generate both flow logs and conversations events
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:validation:Enum:="FLOWS";"CONVERSATIONS";"ENDED_CONVERSATIONS";"ALL"
+	// +kubebuilder:default:=FLOWS
+	LogTypes *string `json:"logTypes,omitempty"`
+
+	//+kubebuilder:default:="30s"
+	// +optional
+	// conversation heartbeat interval is the duration of time to wait between heartbeat reports of a conversation
+	ConversationHeartbeatInterval *metav1.Duration `json:"conversationHeartbeatInterval,omitempty"`
+
+	//+kubebuilder:default:="10s"
+	// +optional
+	// conversation end timeout is the duration of time to wait from the last flow log to end a conversation
+	ConversationEndTimeout *metav1.Duration `json:"conversationEndTimeout,omitempty"`
+
+	// Debug allows setting some aspects of the internal configuration of the flow processor.
+	// This section is aimed exclusively for debugging and fine-grained performance optimizations
+	// (for example GOGC, GOMAXPROCS env vars). Users setting its values do it at their own risk.
+	// +optional
+	Debug DebugConfig `json:"debug,omitempty"`
+}
+
+const (
+	HPAStatusDisabled = "DISABLED"
+	HPAStatusEnabled  = "ENABLED"
+)
+
+type FlowCollectorHPA struct {
+	// +kubebuilder:validation:Enum:=DISABLED;ENABLED
+	// +kubebuilder:default:=DISABLED
+	// Status describe the desired status regarding deploying an horizontal pod autoscaler
+	// DISABLED will not deploy an horizontal pod autoscaler
+	// ENABLED will deploy an horizontal pod autoscaler
+	Status string `json:"status,omitempty"`
+
+	// minReplicas is the lower limit for the number of replicas to which the autoscaler
+	// can scale down.  It defaults to 1 pod.  minReplicas is allowed to be 0 if the
+	// alpha feature gate HPAScaleToZero is enabled and at least one Object or External
+	// metric is configured.  Scaling is active as long as at least one metric value is
+	// available.
+	// +optional
+	MinReplicas *int32 `json:"minReplicas,omitempty" protobuf:"varint,2,opt,name=minReplicas"`
+	// maxReplicas is the upper limit for the number of pods that can be set by the autoscaler; cannot be smaller than MinReplicas.
+	// +kubebuilder:default:=3
+	// +optional
+	MaxReplicas int32 `json:"maxReplicas" protobuf:"varint,3,opt,name=maxReplicas"`
+	// metrics used by the pod autoscaler
+	// +optional
+	Metrics []ascv2.MetricSpec `json:"metrics"`
+}
+
+const (
+	LokiAuthDisabled         = "DISABLED"
+	LokiAuthUseHostToken     = "HOST"
+	LokiAuthForwardUserToken = "FORWARD"
+)
+
+type LokiManualParams struct {
+	//+kubebuilder:default:="http://loki:3100/"
+	// url is the address of an existing Loki service to push the flows to. When using the Loki Operator,
+	// set it to the Loki gateway service with the `network` tenant set in path, for example
+	// https://loki-gateway-http.netobserv.svc:8080/api/logs/v1/network.
+	URL string `json:"url,omitempty"`
```

Comment:
With the Manual configuration we want to take this opportunity to refactor a bit how these URLs are handled, by making them more explicit, with each having a single responsibility. So we already have the querierUrl and the statusUrl, but now they will be required (in manual config) and not optional ; and instead of this general-purpose `URL`, we should have a `IngesterURL` . We'll need also to reformulate the doc text but that can be done in a later step.

Reply:
Looks like this one will require changing FLP. Is that correct? May I proceed with the change and open a second PR there?

Reply:
Hmm I don't see changes necessary in FLP repo (only in `flp_common_objects.go`, from this repository, where you'd need to set

```go
URL:            b.desired.Loki.Manual.IngesterURL,
```

in the `api.WriteLoki` definition

Or do you see something else?

Reply:
Ok. I saw the FLP breaking after the changes. I'll give it a second run and do some troubleshooting on this.

Reply:
So there is actually one change that came in the makefile provoking this error while testing. amd64 is being appended silently to the tags.

Reply:
This one is also done now.